src/java_bytecode/java_bytecode_typecheck_expr.cpp

```diff
@@ -134,28 +136,29 @@ void java_bytecode_typecheckt::typecheck_expr_java_string_literal(exprt &expr)
   const auto &jlo_struct=to_struct_type(ns.follow(jlo_symbol));
   struct_exprt jlo_init(jlo_symbol);
   const auto &jls_struct=to_struct_type(ns.follow(string_type));
+  jlo_init.operands().resize(jlo_struct.components().size());
 
-  jlo_init.copy_to_operands(
-    constant_exprt(
-      "java::java.lang.String",
-      jlo_struct.components()[0].type()));
-  jlo_init.copy_to_operands(
-    from_integer(
-      0,
-      jlo_struct.components()[1].type()));
+  const std::size_t clsid_nb=jlo_struct.component_number("@class_identifier");
```

Comment:
Do we have a function to do JLO init somewhere? If not, we should. The object factory does this too, as does the conversion of `ID_java_new`, and so on and so on.